src/systemd/sd-dhcp-server.h

```diff
@@ -31,7 +31,12 @@ _SD_BEGIN_DECLARATIONS;
 
 typedef struct sd_dhcp_server sd_dhcp_server;
 
-int sd_dhcp_server_new(sd_dhcp_server **ret, int ifindex);
+typedef struct sd_dhcp_server_cb {
+        void (*on_leases_changed)(sd_dhcp_server *s, void *data);
```

Comment:
no open coded structs in public APIs please. let's just have individual setter functions, after allocation for this

Reply:
ack